packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -487,14 +491,24 @@ class CupertinoTextField extends StatefulWidget {
   /// {@macro flutter.services.lengthLimitingTextInputFormatter.maxLength}
   final int? maxLength;
 
-  /// If true, prevents the field from allowing more than [maxLength]
-  /// characters.
-  ///
   /// If [maxLength] is set, [maxLengthEnforced] indicates whether or not to
-  /// enforce the limit, or merely provide a character counter and warning when
-  /// [maxLength] is exceeded.
+  /// enforce the limit.
+  ///
+  /// If false, additional input beyond [maxLength] is ignored.
+  @Deprecated(
+    'Use maxLengthEnforcement parameter which provides more specific '
+    'behavior related to the maxLength limit. '
+    'This feature was deprecated after v1.23.0-19.0.pre.'
+  )
   final bool maxLengthEnforced;
 
+  /// Determines how the [maxLength] limit should be enforced.
+  ///
+  /// Defaults to [MaxLengthEnforcement.enforced].
+  ///
```

Comment:
We need to document what `[MaxLengthEnforcement.none]` does because the warning is not documented anywhere else.